### PR TITLE
mapping: support for False additionalProperties

### PR DESCRIPTION
**Pull request by PXke**
- Handle the case where additionalProperties is False.

Signed-off-by: Guillaume Lastecoueres guillaume@tind.io

### CODE REVIEW

**Comment by PXke** [comment]
All is green :)

**Comment by PXke** [comment]
The reason why I added it :
https://spacetelescope.github.io/understanding-json-schema/reference/object.html#properties

File: domapping/mapping.py [hunk 1]
```diff
@@ -179,14 +179,43 @@ def _gen_type_properties(json_schema, path, resolver, config, es_mapping):
         resolver.push_scope(json_schema.get('id'))
 
     # resolve reference if there are any
+
+    def dict_search_and_retrieve(d, key=None):
+        """
+        Get the values associated to a specific key in nested dicts.
```
Comment (jirikuncar) [inline domapping/mapping.py:185]: Please start the docstring on the same line after `"""`.

``` python
"""Get ...
```

File: domapping/mapping.py [hunk 2]
```diff
@@ -179,14 +179,43 @@ def _gen_type_properties(json_schema, path, resolver, config, es_mapping):
...
+
+        :param d: Dictionnary to explore.
+        :type d: dict
+        :param key: Key to look value for. If None, returns all values.
+        :type key: str
+        :return: A list containing the value associated to Key.
```
Comment (jirikuncar) [inline domapping/mapping.py:191]: `s/Key/key/`

File: domapping/mapping.py [hunk 3]
```diff
@@ -179,14 +179,43 @@ def _gen_type_properties(json_schema, path, resolver, config, es_mapping):
...
 
     # resolve reference if there are any
+
+    def dict_search_and_retrieve(d, key=None):
+        """
+        Get the values associated to a specific key in nested dicts.
+
+        :param d: Dictionnary to explore.
+        :type d: dict
+        :param key: Key to look value for. If None, returns all values.
```
Comment (jirikuncar) [inline domapping/mapping.py:189]:
``` python
if ``None``, it returns ...
```

File: domapping/mapping.py [hunk 4]
```diff
@@ -179,14 +179,43 @@ def _gen_type_properties(json_schema, path, resolver, config, es_mapping):
         resolver.push_scope(json_schema.get('id'))
 
     # resolve reference if there are any
+
+    def dict_search_and_retrieve(d, key=None):
```
Comment (jirikuncar) [inline domapping/mapping.py:183]: If I understand well you are trying to recursively find all values under the `key`. Consider yielding the values so the `any` function can evaluate to `True` early.

Try to find some descriptive name for the function (e.g. `iter_all_values_from_dict(data, key)`, `recursively_iter_all_values(data, key)`, ...) and add some tests.

**Comment by jirikuncar** [comment]
@PXke use topical branches when you create a PR.

**Review by jirikuncar — CHANGES_REQUESTED** [review]
Please refactor the recursive key extraction from a dict. Thanks

## 

see https://github.com/inveniosoftware/domapping/pull/20/files#r78526840

**Pull request closed without merging**